packages/core/src/App/Containers/RealAccountSignup/add-crypto-currency.jsx

```diff
@@ -21,153 +21,138 @@ const Headers = ({ heading, subheading }) => (
     </React.Fragment>
 );
 
-class AddCryptoCurrency extends React.Component {
-    state = {
-        available_fiat_currencies: [],
-    };
-
-    static getDerivedStateFromProps(props) {
-        const fiat = props.legal_allowed_currencies.filter(currency => currency.type === 'fiat');
-        return {
-            available_fiat_currencies: reorderCurrencies(fiat),
-        };
-    }
-
-    get can_add_fiat() {
-        return !this.props.has_fiat && !this.props.should_show_crypto_only;
-    }
+const FIAT_CURRENCY_TYPE = 'fiat';
+const CRYPTO_CURRENCY_TYPE = 'crypto';
 
-    get crypto_currencies() {
-        return reorderCurrencies(
-            this.props.legal_allowed_currencies.filter(currency => currency.type === 'crypto'),
-            'crypto'
+const AddCryptoCurrency = ({
+    legal_allowed_currencies,
+    has_fiat,
+    should_show_crypto_only,
+    should_show_fiat_only,
+    available_crypto_currencies,
+    value,
+    form_error,
+    onSubmit,
```

Comment:
Could we have this list in alphabetical order?

Reply:
Sure